docs/application_architecture.md

```diff
@@ -0,0 +1,104 @@
+## OmiseGO Application Architecture
```

Comment:
call this `OmiseGO Stack Architecture` perhaps? (if so then rename file too). I'd also add a header of 1-2 sentences about what this doc is and what it's not.

On a related topic: this document omits both the child chain and Ethereum, however I think they sort of belong to the story, as being services that this whole stack talks to. Note that such relation can be established in the document without differentiating into what the `child chain` actually is made of - PoA/PoS etc.

Reply:
Will rename and add child chain and Ethereum to the document. Thanks.

Reply:
Under child chain, we should include:
- Fee management: setting of fees, withdrawing fees, possibly burning fees
- Gas management: settings related gas costs for block submissions to Ethereum, process to fund ETH to the operator,

@pdobacz there's more to add here. If you can remember, please add.

Reply:
wait, so my understanding is that this "Stack" doesn't include the child chain, it just "talks" to it. 

So it only should be referenced, but not described, to keep this doc pure and not duplicate with, say, `tesuji_design` or other docs on the child chain server.

The two points mentioned belong outside of this client stack, and will be (already are) described in child chain docs.

Reply:
Ah I get the point. If this is an 'application layer' stack then a reference to the lower layers is appropriate.

Reply:
I'm calling the whole thing "Client services." Open to other suggestions.